server/resp/src/main/java/org/infinispan/server/resp/serialization/Resp3Response.java

```diff
@@ -0,0 +1,231 @@
+package org.infinispan.server.resp.serialization;
+
+import java.util.Collection;
+import java.util.Map;
+import java.util.Set;
+import java.util.function.BiConsumer;
+
+import org.infinispan.server.resp.ByteBufPool;
+
+/**
+ * Entrypoint to serialize a response in RESP3 format.
+ *
+ * <p>
+ * The class provides specific methods for the RESP3 types. In cases where the response object type is unknown or
+ * heterogeneous during runtime, there is a generic method.
+ * </p>
+ *
+ * <p>
+ * During serialization, the class introspects the registry for a serializer capable of handling the object before
+ * proceeding. Utilizing a specific method reduces the number of candidates to check. The generic method verifies all serializers
+ * in the registry and short circuits after the first match. However, this could cause many checks until the correct serializer.
+ * <b>Give preference to specific methods.</b>
+ * </p>
+ *
+ * The method throws an exception when an object does not have a serializer.
+ *
+ * @since 15.0
+ * @author José Bolina
+ */
+public final class Resp3Response {
+
+   private Resp3Response() { }
+
+   public static final BiConsumer<Object, ByteBufPool> OK = (ignore, alloc) -> ok(alloc);
+
+   /**
+    * Serializes a simple string with the <code>"OK"</code> content.
+    *
+    * @param alloc: Buffer pool for allocation.
+    */
+   public static void ok(ByteBufPool alloc) {
+      serialize(RespConstants.OK, alloc, PrimitiveSerializer.SIMPLE_STRING);
+   }
+
+   /**
+    * Serializes a simple string with the <code>"QUEUED"</code> content.
+    *
+    * @param ignore Content to ignore.
+    * @param alloc Buffer pool for allocation.
+    */
+   public static void queued(Object ignore, ByteBufPool alloc) {
+      serialize(RespConstants.QUEUED_REPLY, alloc, PrimitiveSerializer.SIMPLE_STRING);
+   }
+
+   /**
+    * Serializes a char sequence in a simple string format.
+    *
+    * @param value The ASCII string to serialize.
+    * @param alloc Buffer pool for allocation.
+    * @see PrimitiveSerializer#SIMPLE_STRING
+    */
+   public static void simpleString(CharSequence value, ByteBufPool alloc) {
```

Comment:
Explicitly, yeah. But nested objects would very likely utilize simple strings. This is tricky because I didn't find anything explicitly choosing one over the other. Some tests failed because they expected a bulk string, but it was more of a Lettuce thing than for RESP3.